internal/utils/prometheus_transport.go

```diff
@@ -0,0 +1,94 @@
+package utils
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	interfaces "github.com/llm-d-incubation/inferno-autoscaler/internal/interfaces"
+	"github.com/llm-d-incubation/inferno-autoscaler/internal/logger"
+	"github.com/prometheus/client_golang/api"
+)
+
+// CreatePrometheusTransport creates a custom HTTP transport for Prometheus client with TLS support
+func CreatePrometheusTransport(config *interfaces.PrometheusConfig) (http.RoundTripper, error) {
+	// Create base transport
+	transport := &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+		}).DialContext,
+		ForceAttemptHTTP2:     true,
+		MaxIdleConns:          100,
```

Comment:
Do we know why this is set to 100?

Reply:
Controls the maximum number of idle (keep alive) connections to keep per-host, default in Go's http.Transport is 100

Reply:
Can you please add a comment about Go's default.